src/server/services/Actions.ts

```diff
@@ -25,10 +26,17 @@ const rpcTimeout = (service: string) => {
 };
 
 export default class Actions {
+  private static instance: Actions = new Actions();
   account: KeyringPair | undefined;
   #faucetBalance: number | undefined;
 
   constructor() {
+    if (Actions.instance) {
+      throw new Error(
+        'Error: Instantiation failed: Use Actions.getInstance() instead of new.'
```

Comment:
I think `export const actions = new Action()` and making `Action` unexported would be better than this. It would avoid the runtime check and make it less error-prone overall.